### DISCUSSION TITLE
Coverage File Not Being Created

### DISCUSSION THREAD

**Discussion by snayff** [community/community#25477] [Actions]
<p>During the Action the coverage file should be created, which should then be picked up by SonarCloud. Unfortunately, it isnt being created - no error or anything, it just seems to get skipped!</p>
<p>Here’re the relevant blocks in the Action:</p>
<pre><code class="lang-auto">    - name: Pytest with Coverage
      if: always()
      run: |
        poetry run python -m pytest --cov=.

    - name: Push changes
      if: github.repository == 'Snayff/notquiteparadise'
      run: |
        git config --global user.name 'snayff'
        git config --global user.email 'snayff@users.noreply.github.com'
        git commit -am "ci lint and test"
        git push
</code></pre>
<p>Full Action is <a href="https://github.com/Snayff/notquiteparadise/blob/462b3931773406a8d3a3fc5a5189f9af0be6c8d7/.github/workflows/nqp_pr_review.yml" rel="noopener nofollow ugc">here</a>.<br>
Action log is <a href="https://github.com/Snayff/notquiteparadise/runs/1203850758#step:10:1" rel="noopener nofollow ugc">here</a></p>
<p>Any help would be appreciated!</p>

**Comment by airtower-luna** [comment]
<p>Does the coverage file already exist in the repository? <code>git commit -a</code> is not going to add new files, only modified and deleted ones.</p>

**Comment by snayff** [comment]
<p>Ah, no, it doesnt. I (wrongly) thought <code>-am</code> would push all changes, <em>including</em> new files. What should it be instead?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>git add</code> the file, then do a regular commit. That works for new and changed files alike. A <code>git status</code> in between may be useful for debugging, so you can see if the right changes have been staged. :slightly_smiling_face:</p>

**Comment by snayff** [comment]
<p>You’re a star, thank you very much!</p>
